### ISSUE TITLE
run.py incompatible logic in python 3.x

### ISSUE THREAD

**Issue by namachieli** [StackStorm-Exchange/stackstorm-slack#63]
# dictionary changed size during iteration
```
Traceback (most recent call last):
  File \"/opt/stackstorm/st2/lib/python3.6/site-packages/python_runner/python_action_wrapper.py\", line 333, in <module>
    obj.run()
  File \"/opt/stackstorm/st2/lib/python3.6/site-packages/python_runner/python_action_wrapper.py\", line 192, in run
    output = action.run(**self._parameters)
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 19, in run
    return self._do_request(kwargs)
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 32, in _do_request
    for key in params.keys():
RuntimeError: dictionary changed size during iteration
```
https://github.com/StackStorm-Exchange/stackstorm-slack/blob/9aadfdf97201477413555bb7d12ecab9977fc1e4/actions/run.py#L30

Based on the accepted answer in the [Stack overflow discussion:](https://stackoverflow.com/questions/11941817/how-to-avoid-runtimeerror-dictionary-changed-size-during-iteration-error), This needs to be updated to something like `for key in list(params.keys()):`

# 'dict' object has no attribute 'iteritems'
```
Traceback (most recent call last):
  File \"/opt/stackstorm/st2/lib/python3.6/site-packages/python_runner/python_action_wrapper.py\", line 333, in <module>
    obj.run()
  File \"/opt/stackstorm/st2/lib/python3.6/site-packages/python_runner/python_action_wrapper.py\", line 192, in run
    output = action.run(**self._parameters)
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 19, in run
    return self._do_request(kwargs)
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 61, in _do_request
    data = urlencode(encode_obj(params))
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 57, in encode_obj
    return encode_dict(in_obj)
  File \"/opt/stackstorm/packs/slack/actions/run.py\", line 46, in encode_dict
    for k, v in in_dict.iteritems():
AttributeError: 'dict' object has no attribute 'iteritems'
```
https://github.com/StackStorm-Exchange/stackstorm-slack/blob/9aadfdf97201477413555bb7d12ecab9977fc1e4/actions/run.py#L44

In 3.x `iteritems()` was replaced with `items()`, to ensure backwards compatibility a common approach has been to use `six.iteritems(dict)`. In this case; `for k, v in in_dict.iteritems():` should be replaced with `for k, v in six.iteritems(in_dict):`

# Resolution
Checking for other instances of similar logic should be done. Additionally, test coverage would catch this sort of thing, so ideally testing should be created.

**Comment by nmaludy** [comment]
+1 yeah, definitely use `six` here for cross-version compatible coding!

**Comment by namachieli** [comment]
Closing due to fix in #64
